nameko/messaging.py

```diff
@@ -75,22 +74,22 @@ def unpack_message_headers(self, worker_ctx_cls, message):
 
 class Publisher(DependencyProvider, HeaderEncoder):
 
-    def __init__(self, exchange=None, queue=None):
+    def __init__(self, exchange=None, queues_to_declare=None):
```

Comment:
I have added the `queue` argument back with the changes that you suggested to maintain compatibility. I should have included those ones in the first place with my request.

We're now also working with a copy of the original `queues_to_declare` inside the `Publisher` to ensure the provided sequence is a list (so that we can add elements to it later) and to avoid side effects outside this class when adding `queue` to it.

The code now also raises a warning when `queue` is provided. Having a look at other examples in the codebase, I have assumed these changes could be included in version `2.5.2` and also that this warning would be removed in version `2.7.0`. Please let me know if you are happy with this.